src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal abstract class FunctionResolverBase<TProcess, TModule, TRequest>
+        where TProcess : class
+        where TModule : class
+        where TRequest : FunctionResolutionRequestBase<TModule>
+    {
+        internal void EnableResolution(TProcess process, TRequest request)
+        {
```

Comment:
There is also a check in the existing code to ensure we don't try binding causality breakpoints.  We'll want this check also.
```C#
    if (request.CompilerId.LanguageId == DkmLanguageId.MethodId)
    {
        // ignore causality breakpoints
        return;
    }
```